broker.go

```diff
@@ -799,11 +799,17 @@ func (p *producer) Produce(topic string, partition int32, messages ...*proto.Mes
 					"error", err)
 			}
 		}
-		p.conf.Logger.Debug("cannot produce messages",
+		p.conf.Logger.Debug("Cannot produce messages",
 			"retry", retry,
+			"topic", topic,
+			"error", err)
+	}
+	if retry != 0 {
```

Comment:
Also need to consider the case `retry=p.conf.RetryLimit`. Maybe still fail after retry.